test/micro/org/openjdk/bench/java/util/zip/InflaterInputStreams.java

```diff
@@ -108,7 +108,8 @@ public void beforeIteration() throws IOException {
     @Benchmark
     public void inflaterInputStreamRead() throws IOException {
         deflated.reset();
-        InflaterInputStream iis = new InflaterInputStream(deflated);
-        while (iis.read(inflated, 0, inflated.length) != -1);
+        try (InflaterInputStream iis = new InflaterInputStream(deflated)) {
+            while (iis.read(inflated, 0, inflated.length) != -1);
+        }
```

Comment:
Presumably this only works because closing the underlying stream (a BAOS in this case) is a no-op.

Reply:
Yes, the underlying BAIS can be repeatedly closed without affecting the benchmark.

Reply:
Hello Alan, before this change, the memory gets occupied by too many of these:

```
   1:      10871820      521847360  jdk.internal.ref.CleanerImpl$PhantomCleanableRef (java.base@23-internal)
   2:      10871758      260922192  java.util.zip.Inflater$InflaterZStreamRef (java.base@23-internal)

```

Closing the `InflaterInputStream` releases those underlying `InflaterZStreamRef`s.